### PR TITLE
Update 1.20.5

### PR DESCRIPTION
**Pull request by 4drian3d**
Current Supported Version: 1.20.5 Release Candidate 3

### CODE REVIEW

**Comment by AoElite** [comment]
Maybe there could be a option to validate their authentication with cookies if they were transferred from another trusted server. Would reduce calls to mojang's API.

**Comment by 4drian3d** [comment]
> Maybe there could be a option to validate their authentication with cookies if they were transferred from another trusted server. Would reduce calls to mojang's API.

The cookies feature is only for mods environments, the server does not register any cookie by default and if it receives one, you will be immediately kicked out

File: proxy/src/main/java/com/velocitypowered/proxy/protocol/StateRegistry.java [hunk 1]
```diff
@@ -419,11 +450,13 @@ public enum StateRegistry {
           map(0x3D, MINECRAFT_1_19_3, true),
           map(0x41, MINECRAFT_1_19_4, true),
           map(0x43, MINECRAFT_1_20_2, true),
-          map(0x45, MINECRAFT_1_20_3, true));
+          map(0x45, MINECRAFT_1_20_3, true),
+          map(0x46, MINECRAFT_1_20_5, true));
```
Comment (Luccboy) [inline proxy/src/main/java/com/velocitypowered/proxy/protocol/StateRegistry.java]: Shouldn't the respawn packet id be 0x47?

**Review by Luccboy — COMMENTED** [review]
~The SignedChatCommand packet has been split into one that is for commands with signed arguments and one for commands without. This change is currently not implemented, right?~ (already done) Thanks for your great work!

File: proxy/src/main/java/com/velocitypowered/proxy/protocol/StateRegistry.java (hunk 1 above)
Comment (Gerrygames) [reply proxy/src/main/java/com/velocitypowered/proxy/protocol/StateRegistry.java]: good catch!

**Comment by Gerrygames** [comment]
The new unsigned command packet should be handled correctly now

File: proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/ServerLoginSuccessPacket.java [hunk 2]
```diff
@@ -118,6 +121,9 @@ public void encode(ByteBuf buf, ProtocolUtils.Direction direction, ProtocolVersi
         ProtocolUtils.writeProperties(buf, properties);
       }
     }
+    if (version.noLessThan(ProtocolVersion.MINECRAFT_1_20_5)) {
+      buf.writeBoolean(false);
```
Comment (U5B) [inline proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/ServerLoginSuccessPacket.java]: Might want to add a note that this probably will get removed in 1.21
> Changed the approach to game client disconnecting on invalid data in packets sent from a server
> To ease the transition period, modded servers can opt out by setting the appropriate field in the handshake/game_profile packet. This option will be removed in the next release

Comment (4drian3d) [reply proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/ServerLoginSuccessPacket.java]: I don't think a comment about it is so necessary, but I added a system variable to disable that check in the client, by default the check will be enabled

File: proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.velocitypowered.proxy.config.migration;
+
```
Comment (LemonCaramel) [inline proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java:19]:
```suggestion

import com.electronwill.nightconfig.core.CommentedConfig;
```

File: proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+public final class TransferIntegrationMigration implements ConfigurationMigration {
+  @Override
+  public boolean shouldMigrate(final CommentedFileConfig config) {
+    return configVersion(config) < 2.7;
+  }
+
+  @Override
+  public void migrate(final CommentedFileConfig config, final Logger logger) {
+    config.set("accepts-transfers", false);
+    config.setComment("accepts-transfers", """
```
Comment (LemonCaramel) [inline proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java]:
```suggestion
    final CommentedConfig advanced = config.get("advanced");
    advanced.set("accept-transfers", false);
    advanced.setComment("accept-transfers", """
```

Wrongly migration

File: proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java (hunk 3 above)
Comment (LemonCaramel) [inline proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java:19]:
```suggestion

import com.electronwill.nightconfig.core.CommentedConfig;
```

**Review by LemonCaramel — CHANGES_REQUESTED** [review]

File: proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+public final class TransferIntegrationMigration implements ConfigurationMigration {
+  @Override
+  public boolean shouldMigrate(final CommentedFileConfig config) {
+    return configVersion(config) < 2.7;
+  }
+
+  @Override
+  public void migrate(final CommentedFileConfig config, final Logger logger) {
+    config.set("advanced.accepts-transfers", false);
+    config.setComment("advanced.accepts-transfers", """
```
Comment (LemonCaramel) [inline proxy/src/main/java/com/velocitypowered/proxy/config/migration/TransferIntegrationMigration.java:35]: Wrongly migration (2)
```suggestion
    config.set("advanced.accept-transfers", false);
    config.setComment("advanced.accept-transfers", """
```

**Review by kashike — APPROVED** [review]

**Pull request merged**